### PR TITLE
Optimize Ceph log collection and optionally dump to file

### PR DESCRIPTION
**Pull request by fmount**
Instead of running several commands to get the required Ceph logs, this patch improves the way the log tasks are executed. It's now based on a struct that can be iterated.
In addition, it might be a good idea to dump logs, at deployment time, in files, other then (optionally) printing them in the playbook log file.

### CODE REVIEW

**Comment by fultonj** [comment]
+2 looks fine
let's make sure CI is green before merging

File: roles/cifmw_cephadm/tasks/logs.yml
Comment (fmount) [inline roles/cifmw_cephadm/tasks/logs.yml:25]: @fultonj @rebtoor The only potential thing the might result problematic is this path (I'm wondering if we need extra changes to make sure it's collected by the job).
Let's see the first CI run and eventually update the patch.

File: roles/cifmw_cephadm/tasks/logs.yml
Comment (fmount) [reply roles/cifmw_cephadm/tasks/logs.yml:20]: Makes sense, the directory should be already there, we can optimize further by executing this in post when importing `logs.yml` in a block I suppose.

Comment (fmount) [reply roles/cifmw_cephadm/tasks/logs.yml]: done

**Comment by katarimanojk** [comment]
+2 from me, Let us wait for CI.

File: roles/cifmw_cephadm/defaults/main.yml
Comment (rebtoor) [inline roles/cifmw_cephadm/defaults/main.yml:110]:
```suggestion
cifmw_cephadm_log_path: "{{ ansible_user_dir ~ '/ci-framework-data/ceph-logs' }}"
```

File: roles/cifmw_cephadm/tasks/logs.yml
Comment (rebtoor) [inline roles/cifmw_cephadm/tasks/logs.yml:11]: mmh, probably we should improve the renderinof the output here. I was thinking something about:

```suggestion
    msg: |
      'ceph orch status' returned 
      {{ ceph_orch_status.stdout | from_json }}
```

**Comment by fmount** [comment]
```
2024-08-21 14:16:27.193344 | controller | TASK [cifmw_cephadm : Create ceph-logs directory path={{ cifmw_cephadm_log_path }}, state=directory, mode=0755] ***
2024-08-21 14:16:27.193352 | controller | Wednesday 21 August 2024  14:16:27 -0400 (0:00:00.037)       0:31:05.132 ******
2024-08-21 14:16:27.193364 | controller | skipping: [compute-0]
```
... and of course failed because with: `msg: Destination directory /home/zuul/ci-framework-data/ceph-logs does not exist`

**Comment by fmount** [comment]
I see everything is dumped in the the controller node, but the `ceph-logs` directory is not collected during the job report.

**Comment by fmount** [comment]
I can see logs ceph-logs being dumped [1].
I think the patch is now ready, I'll rebase and fix the ceph log json formatting in the next iteration.

[1] https://logserver.rdoproject.org/48/2248/9781a3713fc948f9f62e53ef3c7df6540b35acdc/github-check/podified-multinode-hci-deployment-crc/765b414/controller/ci-framework-data/logs/ceph-logs/

File: roles/cifmw_cephadm/defaults/main.yml
Comment (fmount) [reply roles/cifmw_cephadm/defaults/main.yml]: Done

File: roles/cifmw_cephadm/tasks/logs.yml
Comment (fmount) [reply roles/cifmw_cephadm/tasks/logs.yml]: Done

**Comment by fmount** [comment]
I think this patch is ready to go. We have [1] with a json dump for all the ceph commands executed.
@fultonj @rebtoor let me know if this is ok for you or we need any other improvement.

[1] https://logserver.rdoproject.org/48/2248/5b3737f4543798918273a363fd39d1da5bbde4ee/github-check/podified-multinode-hci-deployment-crc/0954a60/controller/ci-framework-data/logs/ceph/

**Comment by fultonj** [comment]
We also see the `podified-multinode-hci-deployment-crc` is green and we can see the ceph playbook run the new code.

https://logserver.rdoproject.org/48/2248/5b3737f4543798918273a363fd39d1da5bbde4ee/github-check/podified-multinode-hci-deployment-crc/0954a60/controller/ci-framework-data/logs/ansible.log

**Review by fultonj — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
